### PR TITLE
Build using pip with meson backend

### PR DESCRIPTION
**Pull request by ajdawson**
Having seen recent activity on pyspharm, I'm wondering if we can get it into shape and support newer Python versions etc. in the conda-forge channel. A first step here is just attempting to use the meson build backend now used by pyspharm, instead of manually calling setup.py. I have not been able to check my work offline, and I am not sure exactly what changes are needed so I am hoping to use the CI for that. I anticipate further updates to the branch before it is finalised, so there may be a bit of noise from this PR.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,11 +25,12 @@ requirements:
     - {{ stdlib('m2w64_c') }}  # [win]
   host:
     - python
+    - pip
     - numpy
     - meson-python
-    - setuptools <74
   run:
     - python
+    - {{ pin_compatible('numpy') }}
```
Comment (ocefpaf) [inline recipe/meta.yaml:33]: If this works with numpy >=2 we no longer need this. You can declare just numpy here instead.

**Comment by ocefpaf** [comment]
Let's merge this as-is and experiment without the numpy pin in a new PR.

**Comment by ajdawson** [comment]
Ok, this appears to build which is good. I have some more things I'd like to do that can be done in this PR, or others depending on complexity:

* Run the new pyspharm test suite as the build test instead of just importing spharm
* Remove pin on Python < 3.12 to see if this works
* Build for macos arm64 (@ocefpaf do you know how to change the platforms that are built for?)
* Reenable builds on Windows (not sure how hard this is, @ocefpaf I assume it will be good enough if the build/test in CI works, particularly if proper testing is involved as in point 1 above?)

**Comment by ocefpaf** [comment]
> * Build for macos arm64 (@ocefpaf do you know how to change the platforms that are built for?)

We need to send a PR adding it to the list in https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/migrations/osx_arm64.txt, then the bot will send an automated PR adding osx-arm64


> Reenable builds on Windows (not sure how hard this is, @ocefpaf I assume it will be good enough if the build/test in CI works, particularly if proper testing is involved as in point 1 above?)

No idea. We can try flang first. We got a few good results with it recently. Maybe @isuruf has some good recommendations for us.

**Pull request merged**